vault_program/src/update_vault_balance.rs

```diff
@@ -37,23 +37,38 @@ pub fn process_update_vault_balance(
     vault.check_is_paused()?;
 
     // Calculate rewards
-    let new_balance = Account::unpack(&vault_token_account.data.borrow())?.amount;
+    let new_st_balance = Account::unpack(&vault_token_account.data.borrow())?.amount;
 
-    let reward_fee = vault.calculate_rewards_fee(new_balance)?;
+    // 1. Calculate reward fee in VRT
+    let st_rewards = new_st_balance.saturating_sub(vault.tokens_deposited());
+    let st_reward_fee = vault.calculate_st_reward_fee(new_st_balance)?;
 
-    // Update state
-    vault.set_tokens_deposited(new_balance);
-    vault.increment_vrt_supply(reward_fee)?;
+    // 2. Increment ST less the reward fee
+    vault.set_tokens_deposited(new_st_balance.checked_sub(st_reward_fee).unwrap());
```

Comment:
nit: unwrap

Reply:
also im kinda curious what the checked_sub accomplishes compared to just setting new_st_balance?

Reply:
What's happening:

1. We take our fee in st
2. We add the reward ( total reward - fee in st )
3. We virtually call `mint_to` on the reward fee ob behalf of the vault